### ISSUE TITLE
Best way to translate file system path to workspace path without an open project

### ISSUE THREAD

**Issue by norberturkiewicz** [the-carlisle-group/Acre-Desktop#170]
Phil,

API itemname translates with an open project.  What is the best way to translate filepaths to case converted paths or ws path?

```
TestLibrary-11/Activation-1/CreateHashStoreFunction-8441.aplf
TestLibrary-11/Activation-1/IsAuthorized-5.aplf              
TestLibrary-11/Activation-1/RunLicensing-9.aplf              
TestLibrary-11/Activation-1/RunWizard-9.aplf                 
TestLibrary-11/Activation-1/StartLibrary-21.aplf 
```

I need to query function names from other projects to remotely launch tests via command line parameter.  Tests are individual functions in a know parent folder.

**Comment by PhilLast** [comment]
Below is as far as we could take it with the above argument.
```
TestLibrary.Activation.CreateHashStoreFunction 
TestLibrary.Activation.IsAuthorized 
TestLibrary.Activation.RunLicensing 
TestLibrary.Activation.RunWizard 
TestLibrary.Activation.StartLibrary 
```
If the path begins with `aplsource/` or contains `/aplsource/` (case-agnostic and with either "/\\")` we can truncate after that but of course we can't prefix it with the project space as we can't know it without the project's being open.

I'll implement that anyway for any path not recognised as being in a tracked project. It'll be easy to distinguish the result because it won't begin with `#.`.

**Comment by PhilLast** [comment]
I didn't mention it but I mean as an enhancement to `⎕SE.acreAPI.itemname`.

**Comment by norberturkiewicz** [comment]
This was a very dirty and quick way to get my results.  

```
     foos←Directory.GetFiles¨⊂¨folders ⍝↑(⎕NINFO⍠('Recurse' 1)('Wildcard' 1))TestParentPath,'/*.*'
     foos←↑¨⎕NINFO¨↑,/foos
     foos←{('/'≠⍵)⊂⍵}¨(≢appFolder,'/APLSource/')↓¨foos
     foos←{(⍵⍳¨'-')↑¨⍵}¨foos
     foos←(⊂'#.CAS.TestLibrary'),¨∊¨'.',¨¨1↓¨foos
     foos←(~∨/¨foos∊¨⊂'∆')/foos
```

**Comment by norberturkiewicz** [comment]
> Below is as far as we could take it with the above argument.
> 
> ```
> TestLibrary.Activation.CreateHashStoreFunction 
> TestLibrary.Activation.IsAuthorized 
> TestLibrary.Activation.RunLicensing 
> TestLibrary.Activation.RunWizard 
> TestLibrary.Activation.StartLibrary 
> ```
> 
> If the path begins with `aplsource/` or contains `/aplsource/` (case-agnostic and with either "/\")` we can truncate after that but of course we can't prefix it with the project space as we can't know it without the project's being open.
> 
> I'll implement that anyway for any path not recognised as being in a tracked project. It'll be easy to distinguish the result because it won't begin with `#.`.

This works.

**Comment by PhilLast** [comment]
Implemented by commit 5.0.14+179
